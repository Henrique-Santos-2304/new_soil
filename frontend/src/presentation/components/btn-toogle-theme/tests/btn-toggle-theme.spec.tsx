import { fireEvent, screen } from '@testing-library/react'
import { IconToggleTheme } from '../index'
import '@testing-library/jest-dom'
import { renderThemeMock } from '@/tests/index'

describe('Btn Toggler Theme Component', () => {
  it('should render component and write snapshot', () => {
    const { container } = renderThemeMock(<IconToggleTheme />)
    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should expected header to have button container', () => {
    const { container } = renderThemeMock(<IconToggleTheme />)

    const heading = container.querySelector("button[id='btn-theme-container']")

    expect(heading).toBeInTheDocument()
  })
  it('should expected header to have svg of theme light', () => {
    const { container } = renderThemeMock(<IconToggleTheme />)

    const heading = container.querySelector("svg[id='icon-light']")

    expect(heading).toBeInTheDocument()
  })

  it('should expected header to have svg of theme dark when have a click to dark button', () => {
    const { container } = renderThemeMock(<IconToggleTheme />)

    const svgLight = container.querySelector("svg[id='icon-light']")
    svgLight && fireEvent.click(svgLight)
    const svgDark = container.querySelector("svg[id='icon-dark']")
    expect(svgDark).toBeInTheDocument()
  })

  it('should expected header to have heading with change theme text', () => {
    renderThemeMock(<IconToggleTheme />)

    const heading = screen.getByRole('heading', {
      level: 6
    })
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveTextContent('Mudar Tema')
  })
})
