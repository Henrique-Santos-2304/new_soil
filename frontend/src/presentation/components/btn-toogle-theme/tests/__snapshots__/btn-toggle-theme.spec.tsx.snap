// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn Toggler Theme Component should render component and write snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-imz5j4"
  >
    <button
      class="MuiBox-root css-13q49m9"
      id="btn-theme-container"
    >
      <svg
        aria-hidden="true"
        arial-label="botão que muda o tema global do site para claro"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-pnw5j7-MuiSvgIcon-root"
        data-testid="LightModeIcon"
        focusable="false"
        id="icon-light"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41l-1.06-1.06zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
        />
      </svg>
      <h6
        aria-label="Texto indicando que o botão acima é usado para mudar o tema"
        class="MuiTypography-root MuiTypography-h6 css-1khoe0w-MuiTypography-root"
        id="txt-change-theme"
      >
        Mudar Tema
      </h6>
    </button>
  </div>
</div>
`;
